### ISSUE TITLE
Official ducumented support for ajax/XHR/fetch requests

### ISSUE THREAD

**Issue by fractalf** [vuejs/vue#4006]
I just got into vue 2.0 and loving it!

However, when I wanted to do a simple ajax request things kind of stopped...

I *ducked* "vue 2.0 ajax request" and got a lot of different results, non which were *official*.
The most used lib and obvious choise seems to be [vue-resource](https://github.com/vuejs/vue-resource), but there were reports about people not getting in to work etc..

Another "issue" with vue-resource is the "non ES6" promise API, no .catch!
Also it doesn't seem to support the new fetch API.
..I guess these are issues for vue-resource, but I mention them here because that project looks official.

So, my request would be that the official vue 2.0 docs include an example of how to best do an ajax call, so newcomers don't have to spend an hour searching for the best practice on how to do this. I would recon most apps out there use ajax-request, so I think this should be in the docs.

Thanks for a great alternative to react!

**Comment by fnlctrl** [comment]
Hi, thanks for filling this issue.

> so newcomers don't have to spend an hour searching for the best practice 

Best practice: use whatever Ajax/http lib that suits your need, for example, fetch/axios etc or even jquery, and they will do just fine.

The Ajax/http layer should be framework-agnostic, and in fact react doesn't have official http lib either. It is unfortunate that some frameworks like angular/ember left people with the impression that http libs should be bundled with frameworks, and the truth is no, just use whatever you like.

**Comment by LinusBorg** [comment]
And some quick sidenote:
> Another "issue" with vue-resource is the "non ES6" promise API, no .catch!

[Really?](Another "issue" with vue-resource is the "non ES6" promise API, no .catch!)

> Also it doesn't seem to support the new fetch API.

There's [fetch ](https://www.npmjs.com/package/fetch)for that.

> ..I guess these are issues for vue-resource, but I mention them here because that project looks official.

Yeah, vue-resource has some open issues atm, but like vuex, vue-router, vue-validator and other supporting libs we provide, you are free to use them, or use alternatives. There's no lock-in for you, and for many people ans situations, other libs may be better suited than the ones we maintain.

**Comment by fnlctrl** [comment]
> I found superagent, qwest, reqwest and a bunch of others, but yea, which is "best" etc..

A universal way of deciding which lib is better is by comparing their github stars/issues count and npm downlods. ;)

**Comment by LinusBorg** [comment]
And if you aske 3 of our team members which they prefer, you would get three different replies minimum :D

**Comment by posva** [comment]
@fractalf Use Fetch with [fetchival](https://github.com/typicode/fetchival) 😉

**Comment by rajibahmed** [comment]
best-superlative is subjective and always misused in my opinion. Using best makes comparisons feeling like competition.

**Comment by vinny42** [comment]
"The Ajax/http layer should be framework-agnostic"

Why? 

Ajax is an extremely important part of the web these days and you're basically saying "users should find their own way of making our framework work do Ajax". 

The whole point of using a framework is so I *don't* have to figure things out for myself. If I know how to do Ajax, and DOM manipulations etc then why would I use a framework in the first place?

**Comment by LinusBorg** [comment]
> Ajax is an extremely important part of the web these days and you're basically saying "users should find their own way of making our framework work do Ajax".

Not really. What we are saying is:

* Vue is a view-layer library, not a complete framework
* Vue plays nicely with pretty much any AJAX solution out there (axious, superagent, vue-resource, $.ajax...) and integrting them with Vue works pracitcally the ame for all of them: import them into your app and use them as per their README.

We don't see the advantage in importing a Vue-specific AJAX solution - which for many people would be a new API to learn - if the only additional value it offers is that you don't have to install and import it.

In fact, we [retired vue-resource](https://medium.com/the-vue-point/retiring-vue-resource-871a82880af4#.vyiu5frff) for exactly this reason: it offered very littel benefit over using something more mature like $ajax, axios or superagent.

Also, it's about what people need:

* People prefer different solutions, so we don'T want to force one solution on them
* People might not need a general-purpose AJAX lib if they use a Framework or technology that has its own client-side libarary, for example Meteor, FeathersJS, ApolloJS ... 

In those situations, it would be unnessessary bloat to have to include Vue's AJAX implementation into your final bundle, because your don't need it.

**Comment by vinny42** [comment]
"Vue is a view-layer library, not a complete framework"

In that case it makes perfect sense. Thanks for the followup!
